src/Type/Constant/ConstantArrayType.php

```diff
@@ -330,7 +330,12 @@ public function isSuperTypeOf(Type $type): TrinaryLogic
 				} elseif ($hasOffset->maybe() && !$this->isOptionalKey($i)) {
 					$results[] = TrinaryLogic::createMaybe();
 				}
-				$results[] = $this->valueTypes[$i]->isSuperTypeOf($type->getOffsetValueType($keyType));
+
+				$isValueSuperType = $this->valueTypes[$i]->isSuperTypeOf($type->getOffsetValueType($keyType));
+				if ($isValueSuperType->no()) {
+					return TrinaryLogic::createNo();
```

Comment:
```suggestion
					return $isValueSuperType;
```
should be possible, no? or any other reason why you didn't do that?

Reply:
`createNo` is cheap and I like seeing on the first sight without applying any logics what the return type is